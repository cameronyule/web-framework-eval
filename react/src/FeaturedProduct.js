import React from 'react';

const FeaturedProduct = (props) => (
  <div className="featuredproduct">
    <a href={props.href}>
      <p>{props.title}</p>
      <p>{props.description}</p>
    </a>
  </div>
);

FeaturedProduct.propTypes = {
  href: React.PropTypes.string.isRequired,
  title: React.PropTypes.string.isRequired,
  description: React.PropTypes.string.isRequired
}

export default FeaturedProduct;
