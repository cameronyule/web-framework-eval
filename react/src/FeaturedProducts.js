import React from 'react';

import FeaturedProductsList from './FeaturedProductsList';

export default class FeaturedProducts extends React.Component {
  constructor() {
    super();

    this.state = {data: [
      {id: 1, href: "#", title: "Your working week, sorted", linktext: "Shop 9-5 style"},
      {id: 2, href: "#", title: "Half your daily uniform", linktext: "Shop tops"},
      {id: 3, href: "#", title: "Let's dress this bump", linktext: "Shop dresses"},
      {id: 4, href: "#", title: "Layers to lean on", linktext: "Shop maternity essentials"}
    ]};
  }

  render() {
    return (
      <section className="featuredproducts">
        <FeaturedProductsList data={this.state.data} />
      </section>
    )
  }
}
