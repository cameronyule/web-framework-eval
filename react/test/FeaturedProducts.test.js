import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import FeaturedProducts from 'FeaturedProducts';
import FeaturedProduct from 'FeaturedProduct';

describe('FeaturedProducts', () => {
  let products;

  beforeEach(() => {
    products = [
      {id: 1, href: "#", title: "Your working week, sorted", description: "Shop 9-5 style"},
      {id: 2, href: "#", title: "Half your daily uniform", description: "Shop tops"},
      {id: 3, href: "#", title: "Let's dress this bump", description: "Shop dresses"},
      {id: 4, href: "#", title: "Layers to lean on", description: "Shop maternity essentials"}
    ];
  });

  it('renders correctly', () => {
    const wrapper = shallow(<FeaturedProducts data={products} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('requires an array of product data', () => {
    expect(() => {
      console.error = jest.fn();
      const wrapper = shallow(<FeaturedProducts />);
      expect(console.error).toBeCalled();
      expect(console.error.mock.calls[0][0]).toContain('data');
    }).toThrow();
  });

  it('creates a featured product per item', () => {
    const wrapper = shallow(<FeaturedProducts data={products} />);
    expect(wrapper.find(FeaturedProduct).length).toEqual(4);
  });

  it('uses the order defined in the input data array', () => {
    const wrapper = shallow(<FeaturedProducts data={products} />);
    const productsWrapper = wrapper.find(FeaturedProduct);

    const ids = products.map((o) => { return o.id; });
    const productIDs = productsWrapper.map((l) => { return parseInt(l.key()); })

    expect(productIDs).toEqual(ids);
  });
})
