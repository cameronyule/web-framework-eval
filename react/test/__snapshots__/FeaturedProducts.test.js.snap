exports[`FeaturedProducts renders correctly 1`] = `
<section
  className="featuredproducts">
  <FeaturedProduct
    description="Versatility, quality
        and diverse design mean that in AW16’s knitwear collection, there’s a style
        to suit every man no matter his taste or need."
    href="#"
    id={1}
    title="Knit Picks" />
  <FeaturedProduct
    description="A collection of
        products which will enable your life to be lived that little bit more
        stylishly."
    href="#"
    id={2}
    title="Collaborations" />
  <FeaturedProduct
    description="Don’t let your
        autumn/winter wardrobe become a uniform of black, navy and grey. For AW16
        green has received the green light."
    href="#"
    id={3}
    title="The Green Light" />
  <FeaturedProduct
    description="Whether you favour a
        short jacket or a long overcoat, this season’s outerwear is a tale of two
        halves. Discover AW16’s most essential investments."
    href="#"
    id={4}
    title="Coat Tales" />
</section>
`;
